modules/ingest-common/src/main/java/org/opensearch/ingest/common/RemoveByPatternProcessor.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.ingest.common;
+
+import org.opensearch.common.Nullable;
+import org.opensearch.common.ValidationException;
+import org.opensearch.common.regex.Regex;
+import org.opensearch.core.common.Strings;
+import org.opensearch.ingest.AbstractProcessor;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.ingest.IngestDocument;
+import org.opensearch.ingest.Processor;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.opensearch.ingest.ConfigurationUtils.newConfigurationException;
+
+/**
+ * Processor that removes existing fields by field patterns or excluding field patterns.
+ */
+public final class RemoveByPatternProcessor extends AbstractProcessor {
+
+    public static final String TYPE = "remove_by_pattern";
```

Comment:
Wondering why we have this as public since we have a public getter method too, seen this in multiple processors.

Reply:
That's because the getter method is a instance method, not static method, and the static variable `TYPE` is accessed by the external class `IngestCommonModulePlugin`, we cannot access the instance method without constructing a new instance of the processor class.